app/scripts/modules/core/widgets/uiSelect.component.ts

```diff
@@ -0,0 +1,55 @@
+import {Directive, ElementRef, Injector, Input} from '@angular/core';
+import {UpgradeComponent} from '@angular/upgrade/static';
+import {module, IComponentOptions} from 'angular';
+
+class UiSelectComponent implements IComponentOptions {
+
+  public bindings: any = {
+    items: '<',
+    model: '<',
+    modelProperty: '@',
+    placeholder: '@',
+    renderProperty: '@',
+    selectProperty: '@'
+  };
+  public template = `
+    <ui-select ng-model="$ctrl.model[$ctrl.modelProperty]" class="form-control input-sm" required>
+      <ui-select-match placeholder="{{$ctrl.placeholder}}">{{$select.selected[$ctrl.renderProperty]}}</ui-select-match>
+      <ui-select-choices repeat="item[$ctrl.selectProperty] as item in $ctrl.items | filter: $select.search">
+        {{item[$ctrl.renderProperty]}}
```

Comment:
Yea, I don't think we're going to be able to do a "one size fits all" approach with the widget wrappers.  Some, like the helpField component were pretty straight forward.  But given the complexity of ui-select and the options provided, I suspect we'll need several versions.  I realize it's not ideal, but it buys us some time.

I think the next, really big, refactor after pulling the netflix module is going to be moving to bootstrap 4.  Once we do that, we can start using the ng-bootstrap library which has typeahead and will need a little love to get a select option, but shouldn't be scary.  This will allow us to use native ng2 widgets in the ng2 components and not use upgrades and wrappers.

That make sense?